### PR TITLE
DF-3893 - Makefile cleanup + updates

### PR DESCRIPTION
**Pull request by hwilsey-r7**
## Description
- Removed outdated docker files from dockerfiles directory
- Updated makefile to default to Python 3 and build/tag/deploy a single image at a time
- Allow for user input to override the default Python 3 image
- Cleaned up unnecessary python build/installation steps

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,51 +13,31 @@ ifndef TRAVIS_PYTHON_VERSION
...
 
-build:
-	python setup.py build
-
-install:
-	python setup.py install
-
 test:
+	pip install tox
 	tox
```
Comment (zyoutz-r7) [inline Makefile]: Locally this task might make it easier, but when we do the github action I'm of the mindset we do it explicitly: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#running-tests-with-tox. Thoughts?

File: Makefile [hunk 2]
```diff
@@ -13,51 +13,31 @@ ifndef TRAVIS_PYTHON_VERSION
...
 MAKE_VERBOSE=0
 
-image:
-	# Build all 2/3-slim Docker images
-	@for dockerfile in $(wildcard dockerfiles/${DOCKERFILE}*); do \
-		F=$${dockerfile//dockerfiles\/}; \
-		docker build -t komand/python-$${F}-plugin -f dockerfiles/$${F} .; \
-	done
+build-image:
+	docker build -t komand/python-$(DOCKERFILE)-plugin -f dockerfiles/$(DOCKERFILE) .
```
Comment (zyoutz-r7) [inline Makefile]: Add validation that the specific DOCKERFILE exists prior to running these steps

File: Makefile [hunk 3]
```diff
@@ -13,51 +13,31 @@ ifndef TRAVIS_PYTHON_VERSION
...
-	# Tag all 2/3-slim Docker images
-	@for dockerfile in $(wildcard dockerfiles/${DOCKERFILE}*); do \
-		F=$${dockerfile//dockerfiles\/}; \
-		docker tag komand/python-$${F}-plugin komand/python-$${F}-plugin:$(VERSION); \
-		docker tag komand/python-$${F}-plugin komand/python-$${F}-plugin:$(MINOR_VERSION); \
-		docker tag komand/python-$${F}-plugin komand/python-$${F}-plugin:$(MAJOR_VERSION); \
-	done
-
-deploy: tag
+deploy-image:
```
Comment (zyoutz-r7) [inline Makefile]: Call `build-image` to ensure the latest has been built?

Comment (mrinehart-r7) [inline Makefile]:
```suggestion
deploy-image: build-image
```
Agree with @zyoutz-r7

**Comment by mrinehart-r7** [comment]
Might want to kill off the Travis webhooks :)

File: dockerfiles/3-38-slim [hunk 4]
```diff
@@ -12,5 +12,4 @@ ADD ./ /usr/src/komand
 WORKDIR /usr/src/komand
 
 # No `make` being included, so run the python command
-RUN apk add --no-cache openssl==1.1.1b-r1 && \
-	python setup.py install
\ No newline at end of file
+RUN python setup.py install
```
Comment (hwilsey-r7) [inline dockerfiles/3-38-slim]: Removed openssl portion after testing and confirming slim image with plugin action to call HTTPS endpoint.

File: Makefile [hunk 5]
```diff
@@ -13,51 +13,33 @@ ifndef TRAVIS_PYTHON_VERSION
...
+	docker tag komand/python-${DOCKERFILE}-plugin komand/python-${DOCKERFILE}-plugin:$(MINOR_VERSION)
+	docker tag komand/python-${DOCKERFILE}-plugin komand/python-${DOCKERFILE}-plugin:$(MAJOR_VERSION)
 
-all: test tag
-
-build:
-	python setup.py build
+all: install test
 
 install:
```
Comment (zyoutz-r7) [inline Makefile]: I believe this task is for installing the python sdk library locally.  We still want to keep it as is.

File: tests/plugin/hello_world/tests/test_server.py [hunk 6]
```diff
@@ -32,7 +31,7 @@ def plugin_pid():
                     break
                 except Exception as e:
                     pass
-                time.sleep(i / 1000)
+    time.sleep(5)
```
Comment (zyoutz-r7) [inline tests/plugin/hello_world/tests/test_server.py]: Can we add a comment here for more details of why the sleep was implemented

File: Makefile (hunk 5 above)
Comment (hwilsey-r7) [reply Makefile]: Updated

File: tests/plugin/hello_world/tests/test_server.py (hunk 6 above)
Comment (hwilsey-r7) [reply tests/plugin/hello_world/tests/test_server.py]: Added

**Pull request merged**